addon/mixins/base.js

```diff
@@ -1,138 +1,256 @@
 
-      return true;
-    };
+  setSemanticAttr(attrName, attrValue) {
+    return this.execute(`set ${attrName}`, attrValue);
   },
 
-  initializeModule() {
-    this.$()[this.get("module")](this.settings(this.get("module")));
+  areAttrValuesEqual(attrName, attrValue, moduleValue) {
+    return attrValue === moduleValue ||
+           this._stringCompare(attrValue) === moduleValue ||
+           attrValue === this._stringCompare(moduleValue) ||
+           this._stringCompare(attrValue) === this._stringCompare(moduleValue) ||
+           Ember.isEqual(attrValue, moduleValue);
   },
 
-  didInsertElement() {
-    this._super(...arguments);
-    this.initializeModule();
+  // Semantic Helper Methods
+  execute() {
+    let module = this.getSemanticModule();
+    if (module) {
+      return module.apply(this.getSemanticScope(), arguments);
+    }
+  },
 
-    var _this = this;
-    var properties = {};
+  actions: {
+    execute() {
+      return this.execute(...arguments);
+    }
+  },
 
-    // Modules without setable properties
-    properties = this.execute('internal', 'set');
-    var property;
+  // Private Methods
+  _getAttrValue(name) {
+    let value = this.get(`attrs.${name}`);
 
-    if (typeof properties === "object" && !Ember.isArray(properties)) {
-      for(property in properties) {
-        if (!properties.hasOwnProperty(property)) {
-          continue;
-        }
+    if (Ember.isBlank(value)) {
+      return value;
+    }
 
-        if (this.hasOwnProperty(property)) {
-          _this.addObserver(property, _this, _this.updateProperty(property));
-        }
+    // if its a mutable object, get the actual value
+    if (typeof value === 'object') {
+      let objectKeys = Ember.A(Object.keys(value));
+      if (objectKeys.any((objectkey) => objectkey.indexOf('MUTABLE_CELL') === 0)) {
+        value = Ember.get(value, 'value');
       }
     }
+
+    return value;
   },
 
-  willDestroyElement() {
-    var name, selector;
-    if ((selector = this.$()) != null) {
-      if (typeof selector[name = this.get("module")] === "function") {
-        return selector[name]('destroy');
+  _settings() {
+    let moduleName = this.getSemanticModuleName();
+
+    let moduleGlobal = this.getSemanticModuleGlobal();
+    if (!moduleGlobal) {
+      Ember.Logger.error(`Unable to find jQuery Semantic UI module: ${moduleName}`);
+      return;
+    }
+
+    let custom = {
+      debug: Semantic.UI_DEBUG,
+      performance: Semantic.UI_PERFORMANCE,
+      verbose: Semantic.UI_VERBOSE
+    };
+
+    for (let key in this.get('attrs')) {
+      let value = this._getAttrValue(key);
+
+      if (!moduleGlobal.settings.hasOwnProperty(key)) {
+        if (!this.get('_ignorableAttrs').contains(key) && !this.get('_ignorableAttrs').contains(Ember.String.camelize(key))) {
+          // TODO: Add better ember keys here
+          Ember.Logger.debug(`You passed in the property '${key}', but a setting doesn't exist on the Semantic UI module: ${moduleName}`);
```

Comment:
To summarize, there isn't a systematic way I've found to know what properties the base class contains and what properties are passed in. Since when the component is initialized the object hash already contains both. Without coming up with a comprehensive list of all possible html and ember properties, its just a whac a mole game.  During beta I figure we continue to show it, and then drop the logger once we are certain that something isn't being missed